index.d.ts

```diff
@@ -0,0 +1,269 @@
+// Type definitions for JSZip 3.1
+// Project: http://stuk.github.com/jszip/, https://github.com/stuk/jszip
+// Definitions by: mzeiher <https://github.com/mzeiher>, forabi <https://github.com/forabi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="node" />
+
+interface JSZipSupport {
+    arraybuffer: boolean;
+    uint8array: boolean;
+    blob: boolean;
+    nodebuffer: boolean;
+}
+
+type Compression = 'STORE' | 'DEFLATE';
+
+interface Metadata  {
+    percent: number;
+    currentFile: string;
+}
+
+type OnUpdateCallback = (metadata: Metadata) => void;
+
+interface InputByType {
+    base64: string;
+    string: string;
+    text: string;
+    binarystring: string;
+    array: number[];
+    uint8array: Uint8Array;
+    arraybuffer: ArrayBuffer;
+    blob: Blob;
+    stream: NodeJS.ReadableStream;
+}
+
+interface OutputByType {
+    base64: string;
+    text: string;
+    binarystring: string;
+    array: number[];
+    uint8array: Uint8Array;
+    arraybuffer: ArrayBuffer;
+    blob: Blob;
+    nodebuffer: Buffer;
+}
+
+
+interface CompressedObject {
+    compressedSize: number;
+    uncompressedSize: number;
+    crc32: number;
+    compression: object;
+    compressedContent: string|ArrayBuffer|Uint8Array|Buffer;
+}
+
+type InputFileFormat = InputByType[keyof InputByType];
```

Comment:
What does this mean?

Reply:
```suggestion
interface InputByType {
    base64: string;
    string: string;
    text: string;
    binarystring: string;
    array: number[];
    uint8array: Uint8Array;
    arraybuffer: ArrayBuffer;
    blob: Blob;
    stream: NodeJS.ReadableStream;
}
```

InputFileFormat's type is ```string | number[] | Unit8Array |  Blob | NodeJS.ReadableStream```

Reply:
InputFileFormat's type is the element of interface InputByType